Flaky `test_shuffle.py::test_flaky_connect_recover_with_retry`

This test is very flaky:
3 out of 11 runs failed: test_flaky_connect_recover_with_retry (distributed.shuffle.tests.test_shuffle)

e.g. https://github.com/dask/distributed/actions/runs/8156647196/job/22294610970?pr=8551
```
>       assert len(logs) < 600
E       AssertionError: assert 656 < 600
E        +  where 656 = len('Retrying <function ShuffleRun.send.<locals>._send at 0x7fa57efcb4c0> after exception in attempt 0/1: Timed out trying to connect to tcp://127.0.0.1:37919 after 0 s\nRetrying <function ShuffleRun.send.<locals>._send at 0x7fa56eb03160> after exception in attempt 0/1: Timed out trying to connect to tcp://127.0.0.1:37919 after 0 s\nRetrying <function ShuffleRun.send.<locals>._send at 0x7fa57d0ba700> after exception in attempt 0/1: Timed out trying to connect to tcp://127.0.0.1:38667 after 0 s\nRetrying <function ShuffleRun.send.<locals>._send at 0x7fa57d576670> after exception in attempt 0/1: Timed out trying to connect to tcp://127.0.0.1:38667 after 0 s\n')
```

This is a regression introduced in #8511.
CC @hendrikmakait

--- a/distributed/shuffle/tests/test_shuffle.py
+++ b/distributed/shuffle/tests/test_shuffle.py
@@ -2664,8 +2664,10 @@ async def test_flaky_connect_recover_with_retry(c, s, a, b):
         assert rpc.failed_attempts == 1
     # Assert that we do not log the binary payload (or any other excessive amount of data)
     logs = caplog.getvalue()
-    assert len(logs) < 600
-    assert "Retrying" in logs
+    lines = logs.split("\n")
+    for line in lines:
+        assert len(line) < 250
+        assert line.startswith("Retrying")
 
     await check_worker_cleanup(a)
     await check_worker_cleanup(b)

--- a/distributed/shuffle/tests/test_shuffle.py
+++ b/distributed/shuffle/tests/test_shuffle.py
@@ -2667,7 +2667,7 @@ async def test_flaky_connect_recover_with_retry(c, s, a, b):
     lines = logs.split("\n")
     for line in lines:
         assert len(line) < 250
-        assert line.startswith("Retrying")
+        assert not line or line.startswith("Retrying")
 
     await check_worker_cleanup(a)
     await check_worker_cleanup(b)

